main/src/mill/main/MainModule.scala

```diff
@@ -177,4 +182,32 @@ trait MainModule extends mill.Module{
       }
     }
   }
+
+  /**
+    * Deletes the given targets from the out directory. Providing no targets
+    * will clean everything.
+    */
+  def clean(evaluator: Evaluator[Any], targets: String*) = mill.T.command {
+
+    val rootDir = ammonite.ops.pwd / OutDir
+
+    val pathsToRemove =
+      if (targets.isEmpty)
+        Right(List(rootDir))
```

Comment:
In this case mill exits with a non-zero exit code, presumably because it fails to delete `out/clean`, so that's a bit ugly... We should probably list the content of `out/` and filter out `out/clean`.

Reply:
You can get output of clean command with `T.ctx().dest`, and filtering out should be simple in this case